### PR TITLE
tgui-next launchpad

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request
image
image

8 REMAIN

## Changelog
:cl:
add: briefcase launchpad and launchpad console tgui-next
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
The Telescience Guild is pleased with your contribution.

**Comment by kriskog** [comment]
Merge conflict

File: code/game/machinery/launch_pad.dm [hunk 1]
```diff
@@ -349,94 +358,36 @@
...
-					pad.x_offset++
-					pad.update_indicator()
+		if("set_pos")
+			var/new_x = text2num(params["x"])
+			var/new_y = text2num(params["y"])
+			if(!isnull(new_x))
+				pad.set_x(new_x)
+			if(!isnull(new_y))
+				pad.set_y(new_y)
 			. = TRUE
```
Comment (kriskog) [inline code/game/machinery/launch_pad.dm]: You need to call pad.update_indicator()

File: code/game/machinery/launch_pad.dm [hunk 2]
```diff
@@ -349,94 +358,36 @@
...
-					pad.x_offset++
-				pad.update_indicator()
+		if("move_pos")
+			var/plus_x = text2num(params["x"])
+			var/plus_y = text2num(params["y"])
+			if(plus_x)
+				pad.set_x(pad.x_offset + plus_x)
+			if(plus_y)
+				pad.set_y(pad.y_offset + plus_y)
 			. = TRUE
```
Comment (kriskog) [inline code/game/machinery/launch_pad.dm]: Here too

**Review by kriskog — CHANGES_REQUESTED** [review]

File: code/game/machinery/launch_pad.dm (hunk 1 above)
Comment (actioninja) [reply code/game/machinery/launch_pad.dm]: I noticed the indicator wasn't working but just assumed it was broken due to other things lmao

File: tgui-next/packages/tgui/routes.js [hunk 3]
```diff
@@ -44,6 +44,7 @@ import { ImplantChair } from './interfaces/ImplantChair';
 import { KeycardAuth } from './interfaces/KeycardAuth';
 import { LaborClaimConsole } from './interfaces/LaborClaimConsole';
 import { LanguageMenu } from './interfaces/LanguageMenu';
+import { LaunchpadConsole, LaunchpadRemote, LaunchpadControl } from './interfaces/Launchpad';
```
[the remark this answers is not in the document]
Comment (actioninja) [reply tgui-next/packages/tgui/routes.js]: Nope, good catch.

**Review by kriskog — APPROVED** [review]

**Pull request merged**